LICENSE-APACHE

```diff
@@ -0,0 +1,201 @@
+
+6. Trademarks. This License does not grant permission to use the trade
+   names, trademarks, service marks, or product names of the Licensor,
+   except as required for reasonable and customary use in describing the
+   origin of the Work and reproducing the content of the NOTICE file.
+
+7. Disclaimer of Warranty. Unless required by applicable law or
+   agreed to in writing, Licensor provides the Work (and each
+   Contributor provides its Contributions) on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+   implied, including, without limitation, any warranties or conditions
+   of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A
+   PARTICULAR PURPOSE. You are solely responsible for determining the
+   appropriateness of using or redistributing the Work and assume any
+   risks associated with Your exercise of permissions under this License.
+
+8. Limitation of Liability. In no event and under no legal theory,
+   whether in tort (including negligence), contract, or otherwise,
+   unless required by applicable law (such as deliberate and grossly
+   negligent acts) or agreed to in writing, shall any Contributor be
+   liable to You for damages, including any direct, indirect, special,
+   incidental, or consequential damages of any character arising as a
+   result of this License or out of the use or inability to use the
+   Work (including but not limited to damages for loss of goodwill,
+   work stoppage, computer failure or malfunction, or any and all
+   other commercial damages or losses), even if such Contributor
+   has been advised of the possibility of such damages.
+
+9. Accepting Warranty or Additional Liability. While redistributing
+   the Work or Derivative Works thereof, You may choose to offer,
+   and charge a fee for, acceptance of support, warranty, indemnity,
+   or other liability obligations and/or rights consistent with this
+   License. However, in accepting such obligations, You may act only
+   on Your own behalf and on Your sole responsibility, not on behalf
+   of any other Contributor, and only if You agree to indemnify,
+   defend, and hold each Contributor harmless for any liability
+   incurred by, or claims asserted against, such Contributor by reason
+   of your accepting any such warranty or additional liability.
+
+END OF TERMS AND CONDITIONS
+
+APPENDIX: How to apply the Apache License to your work.
+
+   To apply the Apache License to your work, attach the following
+   boilerplate notice, with the fields enclosed by brackets "[]"
+   replaced with your own identifying information. (Don't include
+   the brackets!)  The text should be enclosed in the appropriate
+   comment syntax for the file format. We also recommend that a
+   file or class name and description of purpose be included on the
+   same "printed page" as the copyright notice for easier
+   identification within third-party archives.
+
+Copyright [yyyy] [name of copyright owner]
```

Comment:
We should probably update this placeholders

Reply:
Good point. "yyyy" is always difficult. For now, I'll put 2020-2024 and the name of the copyright owner would be "Contributors to esp-idf-template" as in the MIT license, because we are not asking our contributors to sign a CLA before contributing, so the copyright stays with whoever contributes a PR.

Do you guys in baremetal ask for a CLA assignment? I vaguely remember ESP-IDF wanted that, but we - in Rust land - are running a bit "cowboy" style without is, isn't it?

Maybe that's a topic for Espressif to decide, at least for baremetal.

Reply:
BTW: The `LICENSE-APACHE` and `LICENSE-MIT` in `esp-rs/esp-template` are also just templates.

In `esp-hal` I just noticed the copyright is assigned to `esp-rs` (whatever that means?) and it has out of date copyright dates (2021) should be 2021-2024, as the repo is constantly updated.

Reply:
> Do you guys in baremetal ask for a CLA assignment?

I dont think we have anything in place at the moment.

> BTW: The LICENSE-APACHE and LICENSE-MIT in esp-rs/esp-template are also just templates.

Thanks for the heads up! Ill update them. 

> esp-rs (whatever that means?)  

I guess the org name can be used to be assinged (?)

Reply:
I don't know...? Who is behind the org name? Espressif?